charts/kubewarden-controller/templates/policyserver-default.yaml

```diff
@@ -10,11 +10,23 @@ spec:
   image: {{ .Values.policyServer.image.repository }}:{{ .Values.policyServer.image.tag }}
   serviceAccountName: {{ .Values.policyServer.serviceAccountName }}
   replicas: {{ .Values.policyServer.replicaCount | default 1 }}
-  {{- with .Values.policyServer.annotations }}
   annotations:
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
-  {{- with .Values.policyServer.env }}
+    {{ if .Values.policyServer.telemetry.enabled }}
+      "sidecar.opentelemetry.io/inject": "true"
+    {{ end }}
+    {{- range $key, $value := .Values.policyServer.annotations }}
+      {{ $key | quote }}: {{ $value | quote }}
+    {{- end }}
+  {{ if or .Values.policyServer.env .Values.policyServer.telemetry.enabled }}
   env:
-    {{- toYaml . | nindent 2 }}
+    {{- if .Values.policyServer.telemetry.enabled }}
+    - name: KUBEWARDEN_ENABLE_METRICS
+      value: "1"
+    - name: KUBEWARDEN_LOG_FMT
+      value: otlp
+    {{- end }}
+    {{- range .Values.policyServer.env }}
+    - name: {{ .name | quote }}
+      value: {{ .value | quote }}
+    {{- end }}
```

Comment:
I must admit I don't love the way of enabling opentelemetry on policy servers. I think we have to promote a top level boolean attribute in the `PolicyServer` spec.

Having to:
- Add an annotation (so the opentelemetry webhooks install the opentelemetry collector sidecar)
- Set `KUBEWARDEN_LOG_FMT` to `otlp`
- Set `KUBEWARDEN_ENABLE_METRICS` to something

Is a combination of seemingly unrelated things not trivial to document and hard to remember.

If we add a `.spec.telemetry` boolean flag, this three things would be configured by the controller.

Reply:
that's a good point, and the point you raised about having `.spec.telemetry` be a struct instead of a boolean. Looking forward to it in another iteration.

Reply:
Yes, I totally agree. Please file an issue describing the design of these changes. In the meantime we can stay with this initial proposal